src/main/webapp/3Dscript.js

```diff
@@ -0,0 +1,197 @@
+       * change in our 3D space. Since altitude is already represented in
+       * meters, we simply divide by 4 to adjust for our 1:4 unit ratio.
+       */
+      (point.lng - pose[0].lng) * 25000 * poseTransform.scale,
+      (point.alt - pose[0].alt)/4,
+      (point.lat - pose[0].lat) * 25000 * -poseTransform.scale)
+    );
+  }
+  const geometry = new THREE.BufferGeometry().setFromPoints(coordinates);
+  const material = new THREE.LineBasicMaterial({color: 'blue'});
+  trajectory = new THREE.Line(geometry, material);
+  scene.add(trajectory);
+  trajectory.position.x = poseTransform.translateX;
+  trajectory.position.z = poseTransform.translateZ;
+  trajectory.rotation.y = THREE.Math.degToRad(poseTransform.rotate);
+}
+
+/**
+ * This repositions all children the orientation object. The object 
+ * is a geometry instance, initializing the object with one geometry
+ * call rather than the number of data points. (O(1) vs. O(N))
+ */
+function plotOrientation() {
+  for (point of pose){
+    /**
+     * Matrix is a 4x4 matrix used to translate and rotate each instance
+     * of the pose orientation locally instead using the world transform.
+     * This means it will rotate relative to the position of the matrix, 
+     * not the center of the 3D scene.
+     */ 
+    let matrix = new THREE.Matrix4();
+    matrix.makeTranslation(
+      (point.lng - pose[0].lng) * 25000 * poseTransform.scale,
+      (point.alt - pose[0].alt)/4,
+      (point.lat - pose[0].lat) * 25000 * -poseTransform.scale);
+    matrix.multiply(new THREE.Matrix4().makeRotationX(-Math.PI/2));
+    matrix.multiply(new THREE.Matrix4().makeRotationX(
+        THREE.Math.degToRad(point.pitchDeg)));
+    matrix.multiply(new THREE.Matrix4().makeRotationZ(
+       THREE.Math.degToRad(point.yawDeg)));
+    matrix.multiply(new THREE.Matrix4().makeRotationY(
+        THREE.Math.degToRad(point.rollDeg)));
+    matrix.multiply(new THREE.Matrix4().makeTranslation(0.0, .05, 0.0));
+    orientation.setMatrixAt(i, matrix);
+  }
+  orientation.instanceMatrix.needsUpdate = true; 
+  orientation.position.x = poseTransform.translateX;
+  orientation.position.z = poseTransform.translateZ;
+  orientation.rotation.y = THREE.Math.degToRad(poseTransform.rotate);
+}
+
+/**
+ * This function initializes gui allowing the user to manipulate the
+ * pose objects.
+ */
+function gui() {
+  const gui = new GUI();
+  gui.add(poseTransform, 'rotate', 0, 360,1).onChange(plotOrientation)
+  .onFinishChange(plotTrajectory).name('Pose Rotation (degrees)');
+  gui.add(poseTransform, 'translateX', -10, 10,.025).onChange(plotOrientation)
+  .onFinishChange(plotTrajectory).name('X Axis Translation');
+  gui.add(poseTransform, 'translateZ', -10, 10,.025).onChange(plotOrientation)
+  .onFinishChange(plotTrajectory).name('Z Axis Translation');
+  gui.add(poseTransform, 'scale',.5, 2,.25).onChange(plotOrientation)
+  .onFinishChange(plotTrajectory).name('Pose Scale Multiplier');
```

Comment:
I think this part needs to be formatted; the statement continuations should be indented.